MDANSE/Src/MDANSE/Framework/Jobs/IJob.py

```diff
@@ -186,6 +186,31 @@ def initialize(self):
                     "LineOutputVariable",
                     [index in valid_indices for index in range(array_length)],
                 )
+        self.set_up_trajectory()
+
+    def set_up_trajectory(self):
+        """Apply operations to the trajectory instance, if present.
+
+        Atom selection, atom transmutation and result grouping are all
+        applied to the Trajectory object. If the job works on a trajectory,
+        the Trajectory instance is now saved as an attribute of this IJob
+        instance.
+
+        These operations were previously handled by IConfigurator subclasses.
+        """
+        trajectory = self.configuration.get("trajectory")
+        selection = self.configuration.get("atom_selection")
+        transmutation = self.configuration.get("atom_transmutation")
+        grouping = self.configuration.get("grouping_level")
+        if trajectory is None:
+            return
+        self.trajectory = trajectory["instance"]
+        if selection is not None:
+            self.trajectory.set_selection(selection["flatten_indices"])
+        if transmutation is not None:
+            self.trajectory.set_transmutation(transmutation.transmutation)
+        if grouping is not None:
+            self.trajectory.set_grouping(grouping["level"])
```

Comment:
```suggestion
        if (trajectory := self.configuration.get("trajectory")) is None:
            return
        self.trajectory = trajectory["instance"]
        if (selection := self.configuration.get("atom_selection")) is not None:
            self.trajectory.set_selection(selection["flatten_indices"])
        if (transmutation := self.configuration.get("atom_transmutation")) is not None:
            self.trajectory.set_transmutation(transmutation.transmutation)
        if (grouping := self.configuration.get("grouping_level")) is not None:
            self.trajectory.set_grouping(grouping["level"])
```
Could use walrus here.